src/figuresClasses.ts

```diff
@@ -1,11 +1,92 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green  ' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  readonly shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  readonly shape: Shape;
+
+  constructor(
+    public color: Color,
+    public a: number = 0,
+    public b: number = 0,
+    public c: number = 0,
+  ) {
+    this.shape = 'triangle';
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('some length is <= 0');
+    }
+
+    const sidesValue = [a, b, c];
+    const biggestNum = Math.max(...sidesValue);
+    const otherNumsSum = sidesValue
+      .filter((el) => el !== biggestNum)
+      .reduce((acc, current) => acc + current);
+
+    if (otherNumsSum === biggestNum) {
+      throw new Error(`sides ${a}, ${b} and ${c} can not form a triangle`);
+    }
```

Comment:
Should raise error if `biggestNum is >= otherNumsSum` [Reference](https://github.com/mate-academy/ts_figures_classes#:~:text=the%20longest%20side%20of%20a%20triangle%20is%20%3E%3D%20than%20a%20sum%20of%20two%20others)